### PR TITLE
address and ignore low vulns

### PR DESCRIPTION
**Pull request by daniel-spray**
#### What does this PR do?

- Upgrades from go 1.20 to 1.22
- Upgrades from Alpine 3.18 to 3.20
- Update runAsUser to 10,000 for non openshift helm templates
- Ignores a couple k8s.io package vulns with no fix available

#### Where should the reviewer start?
Go mod and snyk file
#### How should this be manually tested?
Deployed to a testing cluster
#### Any background context you want to provide?
Nope
#### Developer Done List

- [x] Tests Added/Updated
- [x] Updated README.md
- [x] Verified backward compatible
- [x] Verified database migrations will not be catastrophic
- [x] Considered Security, Availability and Confidentiality

#### Reviewer/Approval Done List

- [ ] Tests Pass Locally
- [ ] CI Build Passes
- [ ] Verified README.md is updated
- [ ] Verified changes are backward compatible
- [ ] Reviewed impact to Security, Availability and Confidentiality (if issue found, add comments and request changes)

### CODE REVIEW

File: deploy/docker/Dockerfile [hunk 1]
```diff
@@ -13,7 +13,7 @@ ENV GOOS=linux
 COPY . /go/src/${package}
 RUN go build
 
-FROM alpine:3.18
+FROM alpine:3.20
```
Comment (jdhudson3) [inline deploy/docker/Dockerfile:16]: Any thoughts on just swapping to `3` ?

File: deploy/kubernetes/cloudability-metrics-agent.yaml [hunk 2]
```diff
@@ -113,7 +113,7 @@ spec:
         - image: cloudability/metrics-agent:latest
           imagePullPolicy: Always
           securityContext:
-            runAsUser: 1000
+            runAsUser: 10000
```
Comment (jdhudson3) [inline deploy/kubernetes/cloudability-metrics-agent.yaml:116]: The dockerfile and readme still references user 1000 in multiple places

File: .snyk [hunk 3]
```diff
@@ -35,4 +35,39 @@ ignore:
         reason: No fix available yet
         expires: 2024-10-11T00:00:00.000Z
         created: 2024-07-11T23:07:11.176Z
+  SNYK-GOLANG-K8SIOCLIENTGOTRANSPORT-7538822:
```
Comment (jdhudson3) [inline .snyk:38]: Are these still needed with the 1.22 bump? Seems like the reason should be updated if so.

File: deploy/kubernetes/cloudability-metrics-agent.yaml (hunk 2 above)
Comment (daniel-spray) [reply deploy/kubernetes/cloudability-metrics-agent.yaml:116]: Will update and test again

File: deploy/docker/Dockerfile (hunk 1 above)
Comment (daniel-spray) [reply deploy/docker/Dockerfile:16]: I am fine with this, it will be built with newest alpine version by default in this case

File: .snyk (hunk 3 above)
Comment (daniel-spray) [reply .snyk:38]: Ah I was wrong, it is actually the dependencies fault when upgrading. I updated the error message

File: deploy/kubernetes/cloudability-metrics-agent.yaml (hunk 2 above)
Comment (daniel-spray) [reply deploy/kubernetes/cloudability-metrics-agent.yaml:116]: Things look good after updating those locations

**Review by jdhudson3 — APPROVED** [review]

**Pull request merged**